### PR TITLE
Added FindLoopStartHareTortoise() to check for loops in chapter Question 2_8 

### PR DESCRIPTION
**Pull request by epachayan**
1. Additional function FindLoopStartHareTortoise() included to check for loops in chapter Question 2_8 
2. Related updates in unit tests.
3. Minor typo fix.

### CODE REVIEW

File: CrackingTheCodingInterview/Code/Chapter 2/Question2_8.cs [hunk 1]
```diff
@@ -32,7 +32,47 @@ public static Node<T> FindLoopStart<T>(Node<T> head)
...
+            if (head == null)
+            {
+                throw new ArgumentNullException(nameof(head));
+            }
+
+            Node<T> tortoise = head;
+            Node<T> hare = head;
+
+            // Detect cycle (tortoise and hare approach)
+            while (hare != null && hare.Next != null)
```
Comment (robkeim) [inline CrackingTheCodingInterview/Code/Chapter 2/Question2_8.cs]:
```suggestion
            while (hare?.Next != null)
```

File: CrackingTheCodingInterview/Code/Chapter 2/Question2_8.cs [hunk 2]
```diff
@@ -32,7 +32,47 @@ public static Node<T> FindLoopStart<T>(Node<T> head)
...
+                hare = hare.Next.Next;
+
+                if (ReferenceEquals(tortoise, hare))
+                {
+                    break; // Cycle detected
+                }
+            }
+
+            // No cycle if hare reached the end
+            if (hare == null || hare.Next == null)
```
Comment (robkeim) [inline CrackingTheCodingInterview/Code/Chapter 2/Question2_8.cs]:
```suggestion
            if (hare?.Next == null)
```

File: CrackingTheCodingInterview/Code/Chapter 2/Question2_8.cs [hunk 3]
```diff
@@ -32,7 +32,47 @@ public static Node<T> FindLoopStart<T>(Node<T> head)
             return head;
         }
 
-        private class ReferenceComprer<T> : IEqualityComparer<T>
+        public static Node<T> FindLoopStartHareTortoise<T>(Node<T> head)
```
Comment (robkeim) [inline CrackingTheCodingInterview/Code/Chapter 2/Question2_8.cs:39]: Could you please add the following comments before this method (for consistency with the other methods in the solution):
```
// Space: O(1)
// Time: O(N)
```

**Review by robkeim — CHANGES_REQUESTED** [review]
Thank you very much for your contribution @epachayan!

I've got a few small nits for you and then I'll merge.

File: CrackingTheCodingInterview/Code/Chapter 2/Question2_8.cs (hunk 3 above)
Comment (epachayan) [reply CrackingTheCodingInterview/Code/Chapter 2/Question2_8.cs:39]: raised another PR for the same. Cheers.

**Review by robkeim — APPROVED** [review]

**Comment by robkeim** [comment]
Thank you @epachayan!

**Pull request merged**